_docs/policies/tests.md

```diff
@@ -0,0 +1,94 @@
+# `status-go` Test Policy
+
+- [Creating Tests](#creating-tests)
+- [Flaky Tests](#flaky-tests)
+
+## Creating Tests
+
+- All new functionality MUST be introduced with tests that:
+  - Prove that the functionality performs as described
+  - Can be falsified
+  - Are resistant to fuzzing
+- All new `integration tests` MUST BE validated via 10000 tests.
+  - This can be achieved using the `-count` or `-test.count` flag with the test command eg: `-count 10000` / `-test.count 10000`
+  - Where the CI can not support this work flow automatically, the developer MUST perform validation tests via local testing.
+    - `TODO` Add link to issue for CI automation of validation test runs of new `integration tests`.
+  - Ensuring that the test passes consistently every time gives confidence that the test is not flaky.
+
+## Flaky Tests
+
+Flaky tests are defined as tests that fail intermittently.
+
+- All flaky tests / failing tests MUST be resolved.
+- No flaky tests may be introduced into the codebase.
+
+### Steps to resolving or reporting flaky tests
+
+#### Is it me?
+Determine who caused the flaky test.
+
+- Is a new test you’ve written flaky or failing?
+  - It was you.
+  - You MUST fix the test before merge is acceptable.
+- Has an existing test become flaky?
+  - Check rerun reports. `TODO` add link to rerun reports
+    - If the test does not appear in https://github.com/status-im/status-go/labels/E%3AFlaky%20Test or in the last three nightly test runs, it is most likely that the flakiness was introduced by your changes and needs to be addressed before proceeding with the merge.
+    - Else the test is already documented as a flaky test (appears in the GitHub issues or in the nightly test runs), proceed to below.
+
+```mermaid
+flowchart TB
+    A([PR ready for merge]) --> B{Have any test failed?}
+    B -->|No| C[🎉 Proceed with merge 🪄]
+    B -->|Yes| D{Did you write the test for this PR?}
```

Comment:
> Did you write the test that is failing?

What if you wrote the test 2 years ago? We want to prevent new tests introduced by the PR the developer is trying to merge into `develop`.

Generally if you are introducing code, that code should come with tests. If you change existing code, tests that cover the existing code need to be considered for update.

What about this language?

```suggestion
    B -->|Yes| D{Is the failing test introduced or altered by this PR?}
```